tests/unit/backend/corpora/common/entities/test_project.py

```diff
@@ -84,33 +83,32 @@ def test__create__ok(self):
                 project = Project.create(links=[link_params] * i, **project_params)
 
                 project_key = (project.id, project.status)
-                link_ids = get_ids(project.links)
+                expected_links = project.links
 
-                # Expire all local object and retireve them from the DB to make sure the transactions went through.
+                # Expire all local object and retrieve them from the DB to make sure the transactions went through.
                 Project.db.session.expire_all()
 
-                project = Project.get(project_key)
-                self.assertIsNotNone(project)
-                self.assertEqual(project_key, (project.id, project.status))
-                self.assertEqual(link_ids, get_ids(project.links))
+                project_from_db = Project.get(project_key)
+                self.assertEqual(project_key, (project_from_db.id, project_from_db.status))
+                self.assertCountEqual(expected_links, project_from_db.links)
 
     def test__create_ids__ok(self):
         """
         Creating a project with ids in the links. A new link id is generated even if link id is provided.
         """
         project_params = ProjectParams.get()
-
         project = Project.create(links=[{"id": "test_project_link_id"}], **project_params)
-
         project_key = (project.id, project.status)
-        link_ids = get_ids(project.links)
 
-        # Expire all local object and retireve them from the DB to make sure the transactions went through.
+        # Expire all local object and retrieve them from the DB to make sure the transactions went through.
         Project.db.session.expire_all()
 
-        project = Project.get(project_key)
-        self.assertIsNotNone(project)
-        self.assertEqual(project_key, (project.id, project.status))
+        project_from_db = Project.get(project_key)
+        self.assertEqual(project_key, (project_from_db.id, project_from_db.status))
+        self.assertNotIn("test_project_link_id", project_from_db.links)
```

Comment:
This may seem redundant but this is the only place we are checking that `test_project__link_id` was not linked to the project.